configure/lang/go/lang-go.Deployment.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  annotations:
+    description: Go code intelligence provided by xlang-go, but supporting TLS and
+      WebSockets.
+  labels:
+    deploy: lang-go
+  name: lang-go
+spec:
+  minReadySeconds: 10
+  replicas: 1
+  revisionHistoryLimit: 10
+  selector:
+    matchLabels:
+      app: lang-go
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: lang-go
+    spec:
+      containers:
+      - args:
+        - go-langserver
+        - -mode=websocket
+        - -addr=:7777
+        - -usebuildserver
+        - -usebinarypkgcache=false
+        - -cachedir=$(CACHE_DIR)
+        - -freeosmemory=false
+        env:
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: CACHE_DIR
+          value: /mnt/cache/$(POD_NAME)
```

Comment:
TLS env vars from https://github.com/sourcegraph/deploy-sourcegraph-dot-com/blob/8f7a33406c6d2114abaa55e99a1b4cd849956dfe/configure/lang/go/lang-go.Deployment.yaml#L46-L55

```yaml
        - name: TLS_CERT
          valueFrom:
            secretKeyRef:
              key: cert
              name: tls
        - name: TLS_KEY
          valueFrom:
            secretKeyRef:
              key: key
              name: tls
```

Reply:
@chrismwendt I had to remove them from the default deployment since not everyone will have a secret named `tls` in their system. We should document the existence of these env vars, but they can't be part of the default deployment.